book/optimization.qmd

```diff
@@ -179,152 +242,334 @@ The tuner proceeds as follows:
+  list(cost, gamma, classif.ce, classif.acc)])
 ```
 
 The instance contains a `r ref("BenchmarkResult")` with all `r ref("ResampleResult")`s in `instance$archive$benchmark_result`.
 
 The `r mlr3viz` package provides visualizations for tuning results.
-We plot the performances depending on the evaluated `cp` and `minsplit` values (see. @fig-surface).
+We plot the performances depending on the evaluated `cost` and `gamma` values (see. @fig-surface).
 
 ```{r optimization-016}
 #| label: fig-surface
-#| fig-cap: Performance depending on the evaluated cp and minsplit values.
+#| fig-cap: Performance depending on the evaluated cost and gamma values.
 
 autoplot(instance, type = "surface")
 ```
 
+We observe that high `cost` values and `gamma` values around `exp(-5)` achieve the best performance.
+
 ### Final Model {#sec-final-model}
 
-Now we can fit a final model on the full data set.
-We take the optimized hyperparameters and set them for the previously-created `r ref("Learner")`.
+The learner we use to make predictions on new data is called the final model.
+Let's recap how a model is tuned.
 
 ```{r optimization-017}
-learner$param_set$values = instance$result_learner_param_vals
+learner = lrn("classif.svm",
+  cost  = to_tune(1e-5, 1e5, logscale = TRUE),
+  gamma = to_tune(1e-5, 1e5, logscale = TRUE),
+  kernel = "radial",
+  type = "C-classification"
+)
+
+instance = ti(
+  task = tsk("sonar"),
+  learner = learner,
+  resampling = rsmp("cv", folds = 3),
+  measures = msr("classif.ce"),
+  terminator = trm("evals", n_evals = 20)
+)
+
+tuner = tnr("grid_search", resolution = 5, batch_size = 4)
+
+tuner$optimize(instance)
 ```
 
-We can train the learner on the full dataset.
+After tuning, we can fit a final model on the full data set.
+We take the optimized hyperparameters and set them for the previously-created learner.
 
 ```{r optimization-018}
-learner$train(tsk("penguins"))
+learner$param_set$values = instance$result_learner_param_vals
+```
+
+We train the learner on the full dataset.
+
+```{r optimization-019}
+learner$train(tsk("sonar"))
 ```
 
 The trained model can now be used to predict new, external data.
-The tuning process can also be automated without the need to extract information on the best hyperparameter settings at the end.
-See the section on [automatic tuning](#sec-autotuner).
+The tuning process can also be automated without the need to extract information on the best hyperparameter settings at the end (@sec-autotuner).
 
 :::{.callout-warning}
 A common mistake is to report the performance estimated on the resampling sets on which the tuning was performed (`instance$result$classif.ce`) as the model's performance.
-To get statistically unbiased performance estimates for a given task, [nested resampling](#sec-model-performance) is required.
+To get statistically unbiased performance estimates for a given task, nested resampling (@sec-nested-resampling-unbiased-performance) is required.
 :::
```

Comment:
Please look at the Technical chapter and make sure this doesn't overlap. Either delete content here and just refer to Technical or really simplify this. If you keep it then make sure your example shows an actual error